### DISCUSSION TITLE
Docker Alpine Image

### DISCUSSION THREAD

**Discussion by ghost** [pivpn/pivpn#1676] [Show and tell]
Hi,
I've created an Alpine Docker Image of PiVPN.
I hope you will find it useful.

**Comment by coolapso** [comment]
your effort is great thanks a lot, however I can't recommend anyone to use your image other than yourself. 

https://github.com/giulio-coa/alpine-pivpn/blob/6274f91b20b3f3b5d57fb8f31783f0c9ebc0d7d4/Dockerfile#L18-L20

Everyone using your docker image will have the same certificates and encryption keys, which is a huge no no! 


anyone building their own image based on your Dockerfile should be fine.

**Reply by coolapso** [reply]
however .. don't be discouraged .. its a great progress and quite valuable... some tweaks are needed  most likely on the pivpn script to be able to generate different certificates for each image. 

My vision is that during the startup the docker image checks if data exists on the certificates & configuration directories and act accordingly. 
1 - If data exists, means it has been setup previously, just proceeds with starting up the services. 
2 - If data does not exist then generate all settings and certificates
